generator/wrapper_gen.py

```diff
@@ -177,6 +177,25 @@ def my_procid():
         x = [x.values]
     return new data_or_file(<PyObject *>x)
 
+
+def _str(instance, properties):
+        result = ''
+        for p in properties:
+            prop = getattr(instance, p, None)
+            if isinstance(prop, np.ndarray):
+                result += p + ': array, type={}, shape={}'.format(prop.dtype,
+                                                                  prop.shape)
+            elif isinstance(prop, dict):
+                result += p + ': dict, len={}'.format(len(prop))
+            else:
+                result += p + ': '
+            value = str(prop).replace('\\n', '\\n  ')
```

Comment:
Can you use ``os.linesep`` instead of `\\n`, see https://stackoverflow.com/a/8325958/594376

Reply:
I guess the proper way to do this is the following
```
value = '\n  '.join(str(prop)).splitlines())
```